spacy_llm/cache.py

```diff
@@ -62,7 +65,52 @@ def __init__(
             "persisted": 0,
         }
 
-        self._init_cache_index()
+        self._init_cache_dir()
+
+    @property
+    def prompt_template(self) -> Optional[str]:
+        """Get prompt template.
+        RETURNS (Optional[str]): Prompt template string used for docs to cache/cached docs.
+        """
+        return self._prompt_template
+
+    @prompt_template.setter
+    def prompt_template(self, prompt_template: str) -> None:
+        """Set prompt template.
+        prompt_template (str): Prompt template string used for docs to cache/cached docs.
+        """
+        self._prompt_template = prompt_template
+        if not self._path:
+            return
+
+        # Store prompt template as file.
+        prompt_template_path = self._path / "prompt_template.txt"
+        # If no file exists: store in plain text for easier debugging.
+        if not prompt_template_path.exists():
+            with open(prompt_template_path, "w") as file:
+                file.write(self._prompt_template)
+
+        else:
+            # If file exists and prompt template is not equal to new prompt template: raise, as this indicate we are
+            # trying to reuse a cache directory with a changed prompt.
+            with open(prompt_template_path, "r") as file:
+                existing_prompt_template = "\n".join(file.readlines())
+            if BatchCache._normalize_prompt_template(
+                existing_prompt_template
+            ) != BatchCache._normalize_prompt_template(self._prompt_template):
+                raise ValueError(
+                    f"Prompt template in cache directory ({prompt_template_path}) is not equal with "
+                    f"current prompt template. Reset your cache if you are using a new prompt "
+                    f"template."
+                )
+
+    @staticmethod
+    def _normalize_prompt_template(prompt_template: str) -> str:
```

Comment:
Why not just hash the prompt template here rather than doing this normalization? Whitespace/newlines also matter for what the LLM returns sometimes so if you add some newlines it could easily change the response.

Reply:
Hm, ok. I assumed that whitespace doesn't matter. In that case I didn't want to enforce rebuilding the entire cache just because users added a newline for nicer formatting.

Reply:
Would it make sense to have the prompt be the cache key?

Reply:
I think not, but depends on what you mean specifically. We don't want docs modified via different prompt templates. "Cache key" as I understand it in this context is the hash assigned to a batch/doc bin, which doesn't need the prompt template involved as there should be only one prompt template per cache directory.

Reply:
Updated to hash in https://github.com/explosion/spacy-llm/pull/192/commits/31edb7b9abd47726198523515123cc3151e47fc9.